cpp/oneapi/dal/table/backend/interop/host_soa_table_adapter.cpp

```diff
@@ -0,0 +1,254 @@
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+auto host_soa_table_adapter::deserializeImpl(const daal_dm::OutputDataArchive* arch) -> status_t {
+    return daal::services::ErrorMethodNotImplemented;
+}
+
+void host_soa_table_adapter::freeDataMemoryImpl() {
+    base::freeDataMemoryImpl();
+    original_table_ = homogen_table{};
+}
+
+template <typename BlockData>
+auto host_soa_table_adapter::read_rows_impl(std::size_t vector_idx,
+                                            std::size_t vector_num,
+                                            rw_mode_t rwflag,
+                                            block_desc_t<BlockData>& block) -> status_t {
+    if (rwflag != daal_dm::readOnly) {
+        ONEDAL_ASSERT(!"Data is accessible in read-only mode by design");
+        return daal::services::ErrorMethodNotImplemented;
+    }
+
+    return base::getBlockOfRows(vector_idx, vector_num, rwflag, block);
+}
+
+template <typename BlockData>
+auto host_soa_table_adapter::read_column_values_impl(std::size_t feature_idx,
+                                                     std::size_t vector_idx,
+                                                     std::size_t value_num,
+                                                     rw_mode_t rwflag,
+                                                     block_desc_t<BlockData>& block) -> status_t {
+    if (rwflag != daal_dm::readOnly) {
+        ONEDAL_ASSERT(!"Data is accessible in read-only mode by design");
+        return daal::services::ErrorMethodNotImplemented;
+    }
+
+    return base::getBlockOfColumnValues(feature_idx, vector_idx, value_num, rwflag, block);
+}
+
+bool host_soa_table_adapter::check_row_indexes_in_range(const block_info& info) const {
+    const std::int64_t row_count = original_table_.get_row_count();
+    return info.row_begin_index < row_count && info.row_end_index <= row_count;
+}
+
+bool host_soa_table_adapter::check_column_index_in_range(const block_info& info) const {
+    const std::int64_t column_count = original_table_.get_column_count();
+    return info.single_column_requested && info.column_index < column_count;
+}
+
+template host_soa_table_adapter::host_soa_table_adapter<std::int32_t>(const homogen_table&,
+                                                                      status_t&,
+                                                                      std::int32_t);
+template host_soa_table_adapter::host_soa_table_adapter<float>(const homogen_table&,
+                                                               status_t&,
+                                                               float);
+template host_soa_table_adapter::host_soa_table_adapter<double>(const homogen_table&,
+                                                                status_t&,
+                                                                double);
+
+template auto host_soa_table_adapter::create<std::int32_t>(const homogen_table&) -> ptr_t;
```

Comment:
I think we can remove explicit instantiations for ctors, because we have instantiations for `create`.

Reply:
ctors instantiations were removed